primitives/core/src/bounded/bounded_btree_map.rs

```diff
@@ -168,6 +168,37 @@ where
 	pub fn iter_mut(&mut self) -> sp_std::collections::btree_map::IterMut<K, V> {
 		self.0.iter_mut()
 	}
+
+	/// Consume the map, applying `f` to each of it's values and returning a new map.
+	pub fn map<T, F>(self, mut f: F) -> BoundedBTreeMap<K, T, S>
+	where
+		F: FnMut((&K, V)) -> T,
+	{
+		BoundedBTreeMap::<K, T, S>::unchecked_from(
+			self.0
+				.into_iter()
+				.map(|(k, v)| {
+					let t = f((&k, v));
+					(k, t)
+				})
+				.collect(),
+		)
+	}
+
+	/// Consume the map, applying `f` to each of it's values as long as it returns successfully. If
+	/// an `Err(E)` is ever encountered, the mapping is short circuited and the error is returned;
+	/// otherwise, a new map is returned in the contained `Ok` value.
+	pub fn try_map<T, E, F>(self, mut f: F) -> Result<BoundedBTreeMap<K, T, S>, E>
+	where
+		F: FnMut((&K, V)) -> Result<T, E>,
+	{
+		Ok(BoundedBTreeMap::<K, T, S>::unchecked_from(
+			self.0
+				.into_iter()
+				.map(|(k, v)| (f((&k, v)).map(|t| (k, t))))
+				.collect::<Result<BTreeMap<_, _>, _>>()?,
```

Comment:
Does it really `short circuited` as you mentioned in the documentation?  
To me it looks like it would execute all `f` and then just return an error at the end. So it does not abort after one `f` returned an error.  
Not an issue for me if you make the docs more on-point.

Reply:
Collecting into a Result stops after encountering the first error: https://doc.rust-lang.org/stable/src/core/result.rs.html#2053-2103